src/main/java/com/leahtcodes/Controllers/CoffeeController.java

```diff
@@ -42,13 +50,23 @@ record NewCoffeeRequest(
             Integer coffeePrice,
             String roast
     ){}
+
     @PostMapping("/add")
-    public String addCoffee(@RequestBody NewCoffeeRequest request){
+    public String addCoffee(@RequestBody NewCoffeeRequest request, OAuth2AuthenticationToken authentication){
+        OAuth2AuthorizedClient client = authorizedClientService
+                .loadAuthorizedClient(
+                        authentication.getAuthorizedClientRegistrationId(),
+                        authentication.getName());
+        System.out.println(authentication.getPrincipal().getAttributes());
+        //...
+        Map<String, Object> attributes = authentication.getPrincipal().getAttributes();
+        String id = (String) attributes.get("sub");
```

Comment:
Is this stored as "sub" the subcriber id or something?   You just don't want to hardcode that since it could change, highly unlikely, but possible.